test/integration/utils/edge/config.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package edge
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/kubeedge/kubeedge/test/integration/utils/common"
+)
+
+//config.json decode struct
+type Config struct {
+	MqttEndpoint string `json:"mqtt"`
+	DeviceId     string `json:"deviceId"`
+	TestManager  string `json:"testManager"`
+}
+
+//config struct
+var loadConfig *Config
+
+//get config.json path
+func LoadConfig() Config {
+	if loadConfig == nil {
+		loadConfig = loadConfigJsonFromPath()
+	}
+
+	return *loadConfig
+}
+
+//Load Config.json from the PWD, and decode the config.
+func loadConfigJsonFromPath() *Config {
+	path := getConfigPath()
+	_, err := filepath.Abs(filepath.Dir(path))
+	if err != nil {
+		common.InfoV6("Failed to get Abs path :%v", err)
+	}
+	var config *Config = &Config{}
```

Comment:
you can delay the creation of the config object until line 58. after you read the create the decoder you can do this and pass it to the decode function.